File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    result = [1] * n
    for s in section:
        result[s-1] = 0 
    for i in range(0, n):
        if result[i] == 0:
            for j in range(i, i+m):
                if j < n:
                    result[j] = 1
            answer += 1
    return answer

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.swift
import Foundation

func solution(_ n:Int, _ m:Int, _ section:[Int]) -> Int {
    var count = 0
    var result: [Int] = Array(repeating: 1, count: n)
    for s in section {
        result[s-1] = 0
    }
    for i in 0..<n {
        if result[i] == 0 {
            for j in i..<i+m {
                if j < result.count{
                    result[j] = 1
                }
            }
            count += 1
        }
    }
    return count
}